tensorflow/python/ops/math_ops.py

```diff
@@ -99,9 +99,51 @@
 from tensorflow.python.util.tf_export import tf_export
 
 # Aliases for some automatically-generated names.
-linspace = gen_math_ops.lin_space
+#linspace = gen_math_ops.lin_space
 nextafter = gen_math_ops.next_after
 
+def linspace(start_in, stop_in, num, name=None, axis=0):
+  if not isinstance(axis, int):
+    raise TypeError('Axis should be an integer. Received an object {0} of type {1}'.format(axis, type(axis)))
+  with ops.name_scope(name, 'linspace', [start_in, stop_in]):
+    start = ops.convert_to_tensor(start_in, name='start')
+    # stop must be convertible to the same dtype as start
+    stop = ops.convert_to_tensor(stop_in, name='stop', dtype=start.dtype)
+    expanded_start = array_ops.expand_dims(start, axis=axis)
+    num_int = array_ops.convert_to_int_tensor(num, name='num')
+    num = cast(num_int, dtype=start.dtype)
+    shape = array_ops.shape(expanded_start)
+    ndims = shape.get_shape()[0].value
+    axis = array_ops.get_positive_axis(axis, ndims)
+    expanded_stop = array_ops.expand_dims(stop, axis=axis)
+    # to avoid having negative values in the range or zero division
+    # The result is sliced in the end so a correct result is returned for num == 1.
+    range_end = gen_math_ops.maximum(num - 1., 1.)
+    delta = (expanded_stop - expanded_start) / range_end
+    shape_range = range(ndims)
+    ones_like_shape_range = array_ops.ones_like(shape_range)
+    axis_tiled = ones_like_shape_range * axis
+    # the purpose is to avoid having negative values when repeating
+    num_fill = gen_math_ops.maximum(num_int - 2, 0)
+    num_tiled = ones_like_shape_range * num_fill
+    ones = array_ops.ones_like(num_tiled)
+    mask = gen_math_ops.equal(axis_tiled, shape_range)
+    # reshape_target is [1. 1. 1. ... 1. num 1. 1. ... 1.], where the index of num is equal to axis
+    reshape_target = array_ops.where_v2(mask, num_fill, ones)
+    # repeats is [shape[0], shape[1] ... shape[n-1], 1, shape[n+1], ... shape[-1]] where n is equal to axis
+    repeats = array_ops.where_v2(mask, ones, shape)
+    num_range = range(1., range_end, dtype=start.dtype)
+    range_indices = array_ops.reshape(num_range, reshape_target)
+    tiled_range_indices = array_ops.tile(range_indices, repeats)
+    start_repeated = array_ops.repeat_with_axis(expanded_start, num_fill, axis)
+    delta_repeated = array_ops.repeat_with_axis(delta, num_fill, axis)
+    res = start_repeated + delta_repeated * tiled_range_indices
+    concatenated = array_ops.concat((expanded_start, res, expanded_stop), axis=axis)
+    begin = array_ops.zeros_like(shape)
+    num_slice = ones_like_shape_range * num_int
+    size = array_ops.where_v2(mask, num_slice, shape)
+    return array_ops.slice(concatenated, begin, size)
```

Comment:
There is an edge case here: if we pass `num == -1`, then instead of throwing an exception, we return a concatenated `start` and `stop`. I will check on how this can be validated properly.